handlers/xdelta_handler.c

```diff
@@ -74,28 +171,33 @@ set_default_xd3_config(xd3_config* config, xd3_source* source) {
     config->getblk = NULL; /* We use XD3_GETSRCBLK instead */
 
     // source defaults
-    source->blksize = DEFAULT_GETBLK_SIZE;
+    source->blksize = DEFAULT_BLOCK_SIZE;
+    source->onblk = DEFAULT_BLOCK_SIZE;
+    source->onlastblk = DEFAULT_BLOCK_SIZE;
+    source->max_blkno = DEFAULT_LRU_SIZE - 1;
+    source->max_winsize = DEFAULT_GETBLK_SIZE;
     source->curblk = malloc(source->blksize);
     source->curblkno = (xoff_t) -1;
 }
 
 static size_t
-get_file_size(const char* path) {
-    size_t size = 0;
+get_file_size(char* path) {
+    uint64_t size = 0;
     struct stat sbuf = {0};
-    FILE* f = NULL;
+    int fd;
 
-    if((f = fopen(path, "r")) != NULL) {
-        if(fstat(fileno(f), &sbuf) == 0) {
+    if((fd = open(path, O_RDONLY)) >= 0) {
+        if(fstat(fd, &sbuf) == 0) {
             if (S_ISREG(sbuf.st_mode)) {
                 size = sbuf.st_size;
             } else if (S_ISBLK(sbuf.st_mode)) {
-                ioctl(fileno(f), BLKGETSIZE64, &size);
+                ioctl(fd, BLKGETSIZE64, &size);
             };
         }
-        fclose(f);
+        close(fd);
         return size;
```

Comment:
Is this an intended implicit cast?

Reply:
Yes, ioctl returns 64 bit unsigned int when getting the size of a block device. Since we are running 32bit user space, all types such as `size_t`, `off_t` and `xoff_t` are 32 bit integers.